File: src/Raven.Studio/typescript/commands/database/tasks/getOngoingTasksCommand.ts
import commandBase = require("commands/commandBase");
import database = require("models/resources/database");
import endpoints = require("endpoints");

class getOngoingTasksCommand extends commandBase {

    private readonly db: database;

    private readonly location: databaseLocationSpecifier;

    constructor(db: database, location: databaseLocationSpecifier) {
        super();
        this.location = location;
        this.db = db;
    }

    execute(): JQueryPromise<Raven.Server.Web.System.OngoingTasksResult> {
        const url = endpoints.databases.ongoingTasks.tasks;
        
        const args = {
            ...this.location
        };

        return this.query<Raven.Server.Web.System.OngoingTasksResult>(url, args, this.db);
    }
}

export = getOngoingTasksCommand;

File: src/Raven.Server/Documents/Commands/OngoingTasks/GetOngoingTasksCommand.cs
using System.Net.Http;
using Raven.Client.Http;
using Raven.Server.Web.System;
using Sparrow.Json;

namespace Raven.Server.Documents.Commands.OngoingTasks;

internal class GetOngoingTasksCommand : RavenCommand<OngoingTasksResult>
{
    public GetOngoingTasksCommand(string nodeTag)
    {
        SelectedNodeTag = nodeTag;
    }

    public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
    {
        url = $"{node.Url}/databases/{node.Database}/tasks";

        var request = new HttpRequestMessage { Method = HttpMethod.Get };

        return request;
    }

    public override bool IsReadRequest => true;
}
